packages/backend-app-api/src/services/implementations/auth/authServiceFactory.test.ts

```diff
@@ -224,4 +285,106 @@ describe('authServiceFactory', () => {
+        async (_req, res, ctx) =>
+          res(ctx.json(await catalogAuth.listPublicServiceKeys())),
+      ),
+      rest.get(
+        'http://localhost:7007/api/search/.backstage/auth/v1/jwks.json',
+        async (_req, res, ctx) =>
+          res(ctx.json(await searchAuth.listPublicServiceKeys())),
+      ),
+      rest.get(
+        'http://localhost:7007/api/permission/.backstage/auth/v1/jwks.json',
+        async (_req, res, ctx) =>
+          res(ctx.json(await permissionAuth.listPublicServiceKeys())),
+      ),
+      rest.get(
+        'http://localhost:7007/api/kubernetes/.backstage/auth/v1/jwks.json',
+        (_req, res, ctx) => res(ctx.status(404)),
+      ),
+    );
+
+    const expectedIssuedAt = 1712071714;
+    const expectedExpiresAt = 1712075314;
+
+    jest.useFakeTimers({
+      now: expectedIssuedAt * 1000 + 600_000,
+    });
+
+    const fullToken =
+      'eyJ0eXAiOiJ2bmQuYmFja3N0YWdlLnVzZXIiLCJhbGciOiJFUzI1NiIsImtpZCI6IjhkMDFjM2RiLTU2ZjktNDVmMC04NmRkLTA1YjNjODM1YjNkMyJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjcwMDcvYXBpL2F1dGgiLCJzdWIiOiJ1c2VyOmRldmVsb3BtZW50L2d1ZXN0IiwiZW50IjpbInVzZXI6ZGV2ZWxvcG1lbnQvZ3Vlc3QiLCJncm91cDpkZWZhdWx0L3RlYW0tYSJdLCJhdWQiOiJiYWNrc3RhZ2UiLCJpYXQiOjE3MTIwNzE3MTQsImV4cCI6MTcxMjA3NTMxNCwidWlwIjoiMDFBUUJfSWpHTXRWc2gyWmgzZEg1NXhOX29pSVlhQ1F3ODJjeDZ5M1BQMXlpTjM4eGMzMVpMS2U0YVNDQlJTTy10cjFzZFUzT29ELUxJYV8tNV9RVUEifQ.mjIrZGqbZ2t68fS4U3crlGw-bYJZnMlhMHf-YL7q_u1HfaLr4NMTcHkxdnNS2wfJxCmUBxRfUS8b3nSAKsxcHA';
+
+    const credentials = await searchAuth.authenticate(fullToken);
+    if (!searchAuth.isPrincipal(credentials, 'user')) {
+      throw new Error('not a user principal');
+    }
+    const { token: limitedToken } = await searchAuth.getLimitedUserToken(
+      credentials,
+    );
+
+    const { token: oboToken } = await searchAuth.getPluginRequestToken({
+      onBehalfOf: credentials,
+      targetPluginId: 'catalog',
+    });
+    expect(oboToken).not.toBe(fullToken);
+    expect(decodeJwt(oboToken).obo).toBe(limitedToken);
+    expect(decodeJwt(oboToken).exp).toBe(expectedExpiresAt);
+
+    const oboCredentials = await catalogAuth.authenticate(oboToken);
+    if (!catalogAuth.isPrincipal(oboCredentials, 'user')) {
+      throw new Error('obo credential is not a user principal');
+    }
+    expect(oboCredentials.principal.userEntityRef).toBe(
+      'user:development/guest',
+    );
+    expect(toInternalBackstageCredentials(oboCredentials).token).toBe(
+      limitedToken,
+    );
+
+    const { token: oboToken2 } = await catalogAuth.getPluginRequestToken({
+      onBehalfOf: oboCredentials,
+      targetPluginId: 'permission',
+    });
+    expect(decodeJwt(oboToken2).obo).toBe(limitedToken);
+
+    await expect(
+      catalogAuth.getPluginRequestToken({
+        onBehalfOf: oboCredentials,
+        targetPluginId: 'kubernetes',
+      }),
+    ).rejects.toThrow(
+      "Unable to call 'kubernetes' plugin on behalf of user, because the target plugin does not support on-behalf-of tokens",
```

Comment:
Now I'm curious if this will happen sometimes to users who have a partially migrated plugin flora, just because "modern" plugins in the middle of chains break their stuff

Reply:
Yep idea is to document this dependency in the release notes. I'll start the doc and get it in there. Basically for N>2 deployments one needs to migrate central plugins first. Typically this means you need to migrate permissions before anything else and that's about it. We detect one stop in the chain, but not more.

Reply:
Mkay added a bit of text to the 1.26 notes so we don't forget about this.